### PR TITLE
refactor: use Helm library instead of CLI

### PR DESCRIPTION
**Pull request by f0rmiga**
This refactor focuses on using the Helm library instead of Pods for installing/uninstalling the releases. Also, a small refactor to the listing of k8s resources that affects the Credential reconciliation loop.

### CODE REVIEW

File: operator/helm/helm.go [hunk 1]
```diff
@@ -152,36 +155,62 @@ func (c *Client) Install(name string, chartInfo ChartInfo, opts InstallOpts) err
...
-	res, err := client.Run(name)
+	rel, err := client.Run(name)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get Helm release %q: %w", name, err)
 	}
-	return &Release{res}, nil
+	return rel, nil
+}
+
+// ListResources lists the Kubernetes resources for a given Helm release.
```
Comment (mook-as) [inline operator/helm/helm.go]: Should we say something about the resources here not necessarily matching the server state? (I believe you just return what helm renders, not the resources that are actually deployed.)

File: operator/controllers/instance_controller.go [hunk 2]
```diff
@@ -113,15 +133,15 @@ func (r *InstanceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
...
+		instance.HelmValuesRef.Name = fmt.Sprintf("%s-values-yaml", releaseReq.ReleaseName())
 		instanceNeedsUpdate = true
 	}
 
 	if instanceNeedsUpdate {
 		if err := r.Update(ctx, instance); err != nil {
 			return ctrl.Result{}, err
 		}
-		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+		return ctrl.Result{}, nil
```
Comment (mook-as) [inline operator/controllers/instance_controller.go:144]: No more requeue?

**Review by mook-as — DISMISSED** [review]
We're dropping the provisioning / deprovisioning pods (at least for now), then?

File: operator/controllers/instance_controller.go (hunk 2 above)
Comment (f0rmiga) [reply operator/controllers/instance_controller.go:144]: No - I found out that the `Update` request triggers another event, and the reconcile loop gets triggered again.

File: operator/helm/helm.go (hunk 1 above)
Comment (f0rmiga) [reply operator/helm/helm.go]: Good idea to have a better description. Fixed with eb01c31.

**Comment by f0rmiga** [comment]
> We're dropping the provisioning / deprovisioning pods (at least for now), then?

Correct. Since the `Credential` logic required writing the Helm client, why not use it for everything? The pod strategy is more complex with more objects created and deleted.

**Pull request merged**